go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/ethersphere/bee
 
-go 1.17
+go 1.19
```

Comment:
If go version bump is going to be needed for libp2p upgrade, would it be possible to have effort related go version bump in separate PR without libp2p upgrade? 

I'm aware of draft state, just making heads up comment.

Reply:
a few key dependencies do not compile unless upgraded as well:
`quic-go doesn't build on Go 1.18 yet.`
this triggers a chain of upgrades eventually pulls libp2p v20 or more, so they have to go as a package